pulsar-jms/src/main/java/com/datastax/oss/pulsar/jms/Utils.java

```diff
@@ -457,4 +462,45 @@ static int mapPriorityToPartitionNonLinearly(int priority, int numPartitions) {
     }
     return result;
   }
+
+  /**
+   * Utility to convert path configurations from encoded content into temporary files.
+   * Only configurations that end with "path" (case-insensitive) will be modified.
+   * If the value is a String and begins with "base64:", case-sensitive, it will be base64 decoded then written to a
+   * temp file.
+   * If the value is a byte[], it will be written directly to a temp file.
+   * The temp file permissions will only allow the current user to access the file.
+   * The temp file will be deleted on JVM exit.
+   * @param configuration - the configuration map to modify in place
+   */
+  static void writeEncodedPathConfigsToTempFiles(Map<String, Object> configuration) {
+    for (Map.Entry<String, Object> entry : configuration.entrySet()) {
+      String key = entry.getKey();
+      Object value = entry.getValue();
+      if (key.toLowerCase().endsWith("path")) {
+        try {
+          byte[] bytesToWrite;
+          if (value instanceof String && ((String) value).startsWith("base64:")) {
+            String encoded = ((String) value).replace("\n", "").replace("\r", "").trim();
+            encoded = encoded.substring("base64:".length());
+            bytesToWrite = Base64.getDecoder().decode(encoded);
+          } else if (value instanceof byte[]) {
+            bytesToWrite = (byte[]) value;
+          } else {
+            continue;
+          }
+          Path file = Files.createTempFile("pulsar-jms.", ".tmp");
+          Files.setPosixFilePermissions(file, PosixFilePermissions.fromString("rw-------"));
+          file.toFile().deleteOnExit();
```

Comment:
This is good
I wonder if we could return the list of these files and also eagerly delete them in Pulsar ConnectionFactory.clpse() (as well as keeping this deleteOnExit call)

Reply:
Yes, that definitely makes sense.